From 64b7da9dab82713fdcb2e03d8a2715421eae5685 Mon Sep 17 00:00:00 2001
From: Martin Negyokru <negyokru@inf.u-szeged.hu>
Date: Mon, 3 Oct 2022 12:20:00 +0200
Subject: Do not intercept websocket connection when there is no associated
 frame

This fix is based on chrome's implementation.

Fixes: QTBUG-107144
Pick-to: 6.4
Change-Id: If042e4156b8a4bdb27a210c4db94e3a6198aed7d
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---
 src/core/content_browser_client_qt.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/core/content_browser_client_qt.cpp b/src/core/content_browser_client_qt.cpp
index a1868715c..b6e6748c8 100644
--- a/src/core/content_browser_client_qt.cpp
+++ b/src/core/content_browser_client_qt.cpp
@@ -1245,8 +1245,7 @@ ContentBrowserClientQt::WillCreateURLLoaderRequestInterceptors(content::Navigati
 
 bool ContentBrowserClientQt::WillInterceptWebSocket(content::RenderFrameHost *frame)
 {
-    Q_UNUSED(frame);
-    return true; // It is probably not worth it to only intercept when interceptors are installed
+    return frame != nullptr;
 }
 
 QWebEngineUrlRequestInterceptor *getProfileInterceptorFromFrame(content::RenderFrameHost *frame)
-- 
cgit v1.2.1

